packages/grid/_modules_/grid/hooks/features/export/useCsvExport.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { ApiRef } from '../../../models/api/apiRef';
+import { useApiMethod } from '../../root/useApiMethod';
+import { useGridSelector } from '../core/useGridSelector';
+import { visibleColumnsSelector } from '../columns';
+import { visibleSortedRowsSelector } from '../filter';
+import { selectionStateSelector } from '../selection';
+import { Columns, CsvExportApi, RowId, RowModel } from '../../../models';
+import { useLogger } from '../../utils/useLogger';
+import { exportAs } from '../../../utils';
+
+export const useCsvExport = (apiRef: ApiRef): void => {
+  const logger = useLogger('useCsvExport');
+  const visibleColumns = useGridSelector(apiRef, visibleColumnsSelector);
+  const visibleSortedRows = useGridSelector(apiRef, visibleSortedRowsSelector);
+  const selection = useGridSelector(apiRef, selectionStateSelector);
+
+  const buildRow = (row: RowModel, columns: Columns) => {
+    const mappedRow: RowModel[] = [];
+    columns.forEach((column) => column.field !== '__check__' && mappedRow.push(row[column.field]));
+    return mappedRow;
+  };
+
+  const buildCSV = React.useCallback(
```

Comment:
Does it need to be in this scope? I see the useCallback has no dependencies. Does it mean we can have it in the top level scope of the module?

Reply:
Yes, it does, I added it because TS was complaining. I'll move it out.

Reply:
Moved. I kept it in the same file since I don't force it to be reused somewhere. If we need it we can think of moving it out.